agent/workers/poller.go

```diff
@@ -24,6 +25,8 @@ import (
 	"go.uber.org/zap"
 )
 
+const maxEnvelopeSize = 4 * 1024 * 1024 // 4MB
```

Comment:
I wonder if we should increase this value. What do you think @xoscar, @schoren and @danielbdias ?

Reply:
You mean increasing the value server side too?

Reply:
No, just here. The server is already accepting a large-enough message. Increasing the message here means that in we can send more spans per polling cycle.

Reply:
We discussed and the limit on the server is 4MB, so I'm setting this to be `3MB` to prevent an overflow in case of a large span as discussed in the code.